### PR TITLE
Dma heap changes.

### PR DESCRIPTION
**Pull request by naushir**

### CODE REVIEW

**Comment by naushir** [comment]
@davidplowman

File: core/dma_heaps.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+/*
+ * Copyright (C) 2020, Raspberry Pi Ltd
```
Comment (davidplowman) [inline core/dma_heaps.hpp:3]: 2023?

File: core/dma_heaps.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+/*
+ * Copyright (C) 2020, Raspberry Pi Ltd
```
Comment (davidplowman) [inline core/dma_heaps.cpp:3]: 2023?

File: core/dma_heaps.hpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+#include <libcamera/base/unique_fd.h>
+
+class DmaHeap
+{
+public:
+	DmaHeap();
+	~DmaHeap();
+	bool isValid() const { return dmaHeapHandle_.isValid(); }
+	libcamera::UniqueFD alloc(const char *name, std::size_t size);
```
Comment (davidplowman) [inline core/dma_heaps.hpp:20]: Can this be const? Do we care?

**Comment by davidplowman** [comment]
My other questions notwithstanding, I'm happy merge whenever you want. Perhaps we should also patch up the style checker complaints? Otherwise they might haunt us forever...!

**Pull request merged**